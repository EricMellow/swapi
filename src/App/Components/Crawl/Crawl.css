.crawl {
  width: 100%;
  text-align: center;
  margin: 0 auto;
  font-size: 74px;
  font-family: 'Gothic A1', sans-serif;
  font-weight: 700;
  margin-top: 800px;
  position: relative;
  transform-origin: 50% 100%;
  animation: crawl 60s linear;
}

.crawl-container {
  display: flex;
  justify-content: center;
  width: 100%;
  height: 71vh;
  overflow: hidden;
  perspective: 225px;
  color: #feda4a;
  font-size: 500%;
  letter-spacing: 5px;
  line-height: 150%;
  text-align: justify;
  top: -100px;
}

.title {
  text-align: center;
}

@keyframes crawl {
  0% {
    top: 0;
    transform: rotateX(20deg) translateZ(0);
  }
  100% { 
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px);
  }
}