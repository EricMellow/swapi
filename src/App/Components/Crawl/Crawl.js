import React from 'react';
import './Crawl.css';

const Crawl = (props) => {
  console.log(props)
  const {crawlData} = props;
  let randomNumber = Math.floor(Math.random() * ((6 - 0) +1)) + 0;
  return (
    <div>
      <section className="crawl-container">
        <div className="crawl">
          <div className="title">
            <h1>Episode {crawlData[randomNumber].episode}</h1>
            <h1>{crawlData[randomNumber].title}</h1>
          </div>
          <p>{crawlData[randomNumber].crawl}</p>
          <p>Released {crawlData[randomNumber].release}</p>
        </div>
      </section>
    </div>
  );
};

export default Crawl;