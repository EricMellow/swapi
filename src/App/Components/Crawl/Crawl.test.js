import React from "react";
import Crawl from './Crawl';
import { mockCleanedCrawl } from "../../mockCrawlData.js";
import { shallow } from "enzyme";

describe('Crawl', () => {
  let wrapper;

  beforeEach(() => {
    const mockData = mockCleanedCrawl;
    wrapper = shallow(<Crawl crawlData={mockData}/>);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});