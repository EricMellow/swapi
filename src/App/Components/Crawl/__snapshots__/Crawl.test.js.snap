// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Crawl should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Crawl
    crawlData={
        Array [
            Object {
              "episode_id": 4,
              "opening_crawl": "Opening Crawl Test",
              "release_date": "1977-05-25",
              "title": "A New Hope",
            },
            Object {
              "episode_id": 3,
              "opening_crawl": "Opening Crawl Test",
              "release_date": "1977-05-25",
              "title": "A New Snope",
            },
            Object {
              "episode_id": 3,
              "opening_crawl": "Opening Crawl Test",
              "release_date": "1977-05-25",
              "title": "A New Tommy Cornelius",
            },
            Object {
              "episode_id": 4,
              "opening_crawl": "Opening Crawl Test",
              "release_date": "1977-05-25",
              "title": "A New Hope",
            },
            Object {
              "episode_id": 3,
              "opening_crawl": "Opening Crawl Test",
              "release_date": "1977-05-25",
              "title": "A New Snope",
            },
            Object {
              "episode_id": 3,
              "opening_crawl": "Opening Crawl Test",
              "release_date": "1977-05-25",
              "title": "A New Tommy Cornelius",
            },
            Object {
              "episode_id": 3,
              "opening_crawl": "Opening Crawl Test",
              "release_date": "1977-05-25",
              "title": "A New Tommy Cornelius",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <section
        className="crawl-container"
>
        <div
                className="crawl"
        >
                <div
                        className="title"
                >
                        <h1>
                                Episode 
                        </h1>
                        <h1>
                                A New Tommy Cornelius
                        </h1>
                </div>
                <p
                        className="crawl-text"
                />
                <p
                        className="crawl-text"
                >
                        Released 
                </p>
        </div>
</section>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="crawl"
>
          <div
                    className="title"
          >
                    <h1>
                              Episode 
                    </h1>
                    <h1>
                              A New Tommy Cornelius
                    </h1>
          </div>
          <p
                    className="crawl-text"
          />
          <p
                    className="crawl-text"
          >
                    Released 
          </p>
</div>,
        "className": "crawl-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="title"
>
              <h1>
                            Episode 
              </h1>
              <h1>
                            A New Tommy Cornelius
              </h1>
</div>,
            <p
              className="crawl-text"
/>,
            <p
              className="crawl-text"
>
              Released 
</p>,
          ],
          "className": "crawl",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  Episode 
</h1>,
                <h1>
                  A New Tommy Cornelius
</h1>,
              ],
              "className": "title",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Episode ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Episode ",
                  undefined,
                ],
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "A New Tommy Cornelius",
                },
                "ref": null,
                "rendered": "A New Tommy Cornelius",
                "type": "h1",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "crawl-text",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Released ",
                undefined,
              ],
              "className": "crawl-text",
            },
            "ref": null,
            "rendered": Array [
              "Released ",
              undefined,
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": "section",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section
          className="crawl-container"
>
          <div
                    className="crawl"
          >
                    <div
                              className="title"
                    >
                              <h1>
                                        Episode 
                              </h1>
                              <h1>
                                        A New Tommy Cornelius
                              </h1>
                    </div>
                    <p
                              className="crawl-text"
                    />
                    <p
                              className="crawl-text"
                    >
                              Released 
                    </p>
          </div>
</section>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="crawl"
>
            <div
                        className="title"
            >
                        <h1>
                                    Episode 
                        </h1>
                        <h1>
                                    A New Tommy Cornelius
                        </h1>
            </div>
            <p
                        className="crawl-text"
            />
            <p
                        className="crawl-text"
            >
                        Released 
            </p>
</div>,
          "className": "crawl-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="title"
>
                <h1>
                                Episode 
                </h1>
                <h1>
                                A New Tommy Cornelius
                </h1>
</div>,
              <p
                className="crawl-text"
/>,
              <p
                className="crawl-text"
>
                Released 
</p>,
            ],
            "className": "crawl",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1>
                    Episode 
</h1>,
                  <h1>
                    A New Tommy Cornelius
</h1>,
                ],
                "className": "title",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Episode ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Episode ",
                    undefined,
                  ],
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "A New Tommy Cornelius",
                  },
                  "ref": null,
                  "rendered": "A New Tommy Cornelius",
                  "type": "h1",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "crawl-text",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Released ",
                  undefined,
                ],
                "className": "crawl-text",
              },
              "ref": null,
              "rendered": Array [
                "Released ",
                undefined,
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        "type": "section",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
